packages/-ember-data/addon/-private/system/debug/debug-adapter.js

```diff
@@ -18,6 +19,16 @@ import Model from '@ember-data/model';
   @private
 */
 export default DataAdapter.extend({
+  store: service('store'),
+
+  init() {
+    this._super(...arguments);
+    let owner = getOwner(this.get('store'));
+    if (!owner.hasRegistration('data-adapter:main')) {
```

Comment:
I believe this is too late in time. I may be wrong but I think files at the top level are considered "main", so exporting this as `app/debug-adapter.js` should "work". @rwjblue can you confirm?

Reply:
I believe you, but can you explain (🤞 with links) on how it "work" so I can understand?

Reply:
yes, `app/debug-adapter.js` would be what is looked for when you do `owner.lookup('debug-adapter:main')`.